### ISSUE TITLE
The performance benchmark seems not good.

### ISSUE THREAD

**Issue by upeoe** [SkyAPM/SkyAPM-php-sdk#485]
**Describe the bug**
As we use the v3.x at the production environment, and want to upgrade the extension to the v4.x, so we had a stress testing. 
But the result is not good. Is there has official benchmark testing to point out that maybe our testing is incorrect?

**To Reproduce**
1. Writes a health check api with just echo the response `ok`.
2. Build the extension `so` binary file following the guideline.

**PHP :**
``` bash
$ php -v
PHP 7.2.34 (cli) (built: Sep 15 2021 00:24:42) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.34, Copyright (c) 1999-2018, by Zend Technologies

# And the php-fpm worker numbers are 30.
```

**OS:**
``` bash
$ cat /etc/redhat-release
CentOS Linux release 7.6.1810 (Core)

# And the OS is 4c8g.
```

**SkyWalking Server:**
 - OS: CentOS
 - Version: 8

**Benchmark Testing Report**

***tldr***:

1. The QPS:
![image](https://user-images.githubusercontent.com/11611673/165926032-a703f3e4-a295-4be6-9b15-acc2e7bb9e35.png)

2. The P99 latency:

![image](https://user-images.githubusercontent.com/11611673/165925739-ece7590d-fc89-4182-8703-5432a1c9120c.png)

**The extension v3.3.2**

1. Without tracing:

``` bash
$ php --ri skywalking
skywalking.enable => Off => Off

$ wrk --latency -d60s -c100 http://${client_endpoint}/health
Running 1m test @ http://${client_endpoint}/health
  2 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    90.17ms    4.75ms 216.32ms   84.55%
    Req/Sec   556.54     30.05   660.00     70.64%
  Latency Distribution
     50%   89.52ms
     75%   91.92ms
     90%   94.96ms
     99%  105.26ms
  66532 requests in 1.00m, 40.67MB read
Requests/sec:   1107.31
Transfer/sec:    693.10KB
```

2. Full open tracing:

``` bash
$ php --ri skywalking
skywalking.enable => On => On

$ wrk --latency -d60s -c100 http://${client_endpoint}/health
Running 1m test @ http://${client_endpoint}/health
  2 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   108.16ms    5.25ms 265.57ms   85.35%
    Req/Sec   463.87     27.09   515.00     83.33%
  Latency Distribution
     50%  107.47ms
     75%  109.96ms
     90%  112.97ms
     99%  123.74ms
  55490 requests in 1.00m, 33.92MB read
Requests/sec:    923.63
Transfer/sec:    578.13KB
```

**The extension v4.2.0**

1. Without tracing:

``` bash
$ php --ri skywalking
skywalking.enable => Off => Off
skywalking.sample_n_per_3_secs => -1 => -1

$ wrk --latency -d60s -c100 http://${client_endpoint}/health
Running 1m test @ http://${client_endpoint}/health
  2 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    89.95ms    3.92ms 221.23ms   87.38%
    Req/Sec   558.02     29.20   630.00     82.15%
  Latency Distribution
     50%   89.65ms
     75%   91.35ms
     90%   93.14ms
     99%   99.35ms
  66714 requests in 1.00m, 40.78MB read
Requests/sec:   1110.32
Transfer/sec:    694.98KB
```

2. The sample of 1 per 3 seconds:

``` bash
$ php --ri skywalking
skywalking.enable => On => On
skywalking.sample_n_per_3_secs => 1 => 1

$ wrk --latency -d60s -c100 http://${client_endpoint}/health
Running 1m test @ http://${client_endpoint}/health
  2 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   149.75ms    7.93ms 322.83ms   88.60%
    Req/Sec   335.19     37.70   470.00     78.45%
  Latency Distribution
     50%  148.76ms
     75%  152.01ms
     90%  155.98ms
     99%  171.92ms
  40029 requests in 1.00m, 24.47MB read
Requests/sec:    666.83
Transfer/sec:    417.39KB
```

3. The sample of 10 per seconds:

``` bash
$ php --ri skywalking
skywalking.enable => On => On
skywalking.sample_n_per_3_secs => 1 => 10

$ wrk --latency -d60s -c100 http://${client_endpoint}/health
Running 1m test @ http://${client_endpoint}/health
  2 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   150.06ms    8.91ms 328.86ms   84.94%
    Req/Sec   334.30     31.67   474.00     76.38%
  Latency Distribution
     50%  148.53ms
     75%  153.33ms
     90%  158.67ms
     99%  179.17ms
  39960 requests in 1.00m, 24.43MB read
Requests/sec:    664.98
Transfer/sec:    416.23KB
```

4. The sample of 100 per seconds:

``` bash
$ php --ri skywalking
skywalking.enable => On => On
skywalking.sample_n_per_3_secs => 1 => 100

$ wrk --latency -d60s -c100 http://${client_endpoint}/health
Running 1m test @ http://${client_endpoint}/health
  2 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   157.58ms    6.94ms 240.44ms   85.91%
    Req/Sec   318.26     62.36   505.00     71.92%
  Latency Distribution
     50%  157.30ms
     75%  160.45ms
     90%  163.73ms
     99%  174.71ms
  38065 requests in 1.00m, 23.27MB read
Requests/sec:    633.72
Transfer/sec:    396.67KB
```

5. Full open tracing:

``` bash
$ php --ri skywalking
skywalking.enable => On => On
skywalking.sample_n_per_3_secs => -1 => -1

$ wrk --latency -d60s -c100 http://${client_endpoint}/health
Running 1m test @ http://${client_endpoint}/health
  2 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   157.80ms    7.87ms 255.76ms   86.05%
    Req/Sec   317.87     36.42   470.00     79.47%
  Latency Distribution
     50%  157.06ms
     75%  160.75ms
     90%  164.85ms
     99%  184.31ms
  37981 requests in 1.00m, 23.22MB read
Requests/sec:    632.79
Transfer/sec:    396.09KB
```
